bin/gce-pre-align.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+set -o pipefail
+
+ZONES[0]="us-central1-a"
+ZONES[1]="us-central1-b"
+ZONES[2]="us-central1-c"
+ZONES[3]="us-central1-f"
+MACHINE_ZONE=${ZONES[$[ $RANDOM % 4 ]]}
+MACHINE_NAME=$(echo "pre-align-"$(basename $1 | cut -f 1 -d '.') | tr "[:upper:]" "[:lower:]" | sed "s/[^a-z0-9]/-/g" | head -c62)
+#MACHINE_TYPE_OPTS="--machine-type n1-standard-2"
+MACHINE_TYPE_OPTS="--custom-cpu 1 --custom-memory 6656MiB"
+
+gcloud compute instances create --zone $MACHINE_ZONE --scopes storage-full --image ubuntu-1404-alignment $MACHINE_TYPE_OPTS --boot-disk-size 300 $MACHINE_NAME
+
+EXIT_STATUS=$?
+if [[ $EXIT_STATUS == 0 ]]
+then
+  # Give ssh daemon some time to get running.
+  sleep 30s  
+
+  echo "[$(date)] Downloading input crams"
+  START_TIME=$(date +%s)
+  INPUT_FILE="/home/alignment/"$(basename $1)
+  OUT_BASE="/home/alignment/"$(basename $1 | cut -f 1 -d '.')
+ 
+  gcloud compute ssh --zone $MACHINE_ZONE $MACHINE_NAME -- gsutil cp $1 $INPUT_FILE
+  EXIT_STATUS=$?
+  echo "[$(date)] Elapsed time: "$(( $(date +%s) - $START_TIME ))"s"
+
+  if [[ $EXIT_STATUS == 0 ]]
+  then
+    REMOTE_COMMAND="set -euo pipefail &&
+      export REF_PATH=/home/alignment/ref/md5/%2s/%2s/%s &&
+      samtools view -uh -F 0x900 $INPUT_FILE \
+      | bam-ext-mem-sort-manager squeeze --in -.ubam --keepDups --rmTags AS:i,BD:Z,BI:Z,XS:i,MC:Z,MD:Z,NM:i,MQ:i --out -.ubam \
+      | samtools sort -l 1 -@ 1 -m 4000M -n -T /home/alignment/sort_tmp - \
+      | samtools fixmate - - \
+      | bam-ext-mem-sort-manager bam2fastq --in -.bam --outBase $OUT_BASE --maxRecordLimitPerFq 20000000 --sortByReadNameOnTheFly --readname --gzip 2> /home/alignment/fastq_err.log
+      fastq_reads=\\\$((\\\$(zcat /home/alignment/*.fastq.gz | wc -l) / 4))
+      samtools flagstat $INPUT_FILE > /home/alignment/cram_flagstat.txt
```

Comment:
The next time reads are checked, we will have already downloaded the recalibrated cram back to csg and we already have the read count stored in DB.